detox/ios/Detox/DetoxManager.m

```diff
@@ -90,11 +90,14 @@ - (instancetype)init
 	
 	[[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(_appDidLaunch:) name:UIApplicationDidFinishLaunchingNotification object:nil];
 	[[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(_appDidEnterBackground:) name:UIApplicationDidEnterBackgroundNotification object:nil];
-	
-	NSString* recordingPath = [NSUserDefaults.standardUserDefaults objectForKey:@"recordingPath"];
+
+	NSUserDefaults *userDefaults = NSUserDefaults.standardUserDefaults;
+	NSString* recordingPath = [userDefaults objectForKey:@"recordingPath"];
 	if(recordingPath != nil)
 	{
-		[self _handlePerformanceRecording:NSDictionaryOfVariableBindings(recordingPath) isFromLaunch:YES completionHandler:nil];
+		NSString* samplingInterval = [userDefaults objectForKey:@"samplingInterval"];
```

Comment:
Not sure having "samplingInterval" as the launch argument is a good idea, it's too generic. This is also true about "recordingPath".

Reply:
Are you talking about naming problem or about the idea of passing launch arguments as general idea?

Reply:
I mean naming. There is no escape from launch arguments for now because they have to be passed very early on so that recording starts as early as possible. But "recordingPath" and "samplingInterval" are quite generic, so there may be collision with user arguments.